### PR TITLE
Upgrades to Scala 2.11

### PR DESCRIPTION
**Pull request by stephanh**
Fixes a whole bunch of compile errors and warnings in the process.

### CODE REVIEW

File: core/src/main/scala/au/com/cba/omnia/omnitool/Closeable.scala [hunk 1]
```diff
@@ -20,7 +20,10 @@ case class Closeable[T](close: T => Any)
 
 case class CloseableSyntax[T](v: T, closeable: Closeable[T]) {
   /** Close the resource */
-  def close(): Unit = closeable.close(v)
+  def close(): Unit = {
+    closeable.close(v)
+    ()
```
Comment (vineethvarghese) [inline core/src/main/scala/au/com/cba/omnia/omnitool/Closeable.scala]: Do we at any point expect a `Closeable` to return a value? Why can't it be a side effect only operation?

File: core/src/test/scala/au/com/cba/omnia/omnitool/ValidatedSpec.scala [hunk 2]
```diff
@@ -23,8 +23,9 @@ Error handling
 ==============
 
 Validated:
-  can be used to wrap operations that might throw exceptions $safe
-  allow for adding additional error messages to already existing errors $addError
+  can be used to wrap operations that might throw exceptions safe
+  allow for adding additional error messages to already existing errors addError
```
Comment (vineethvarghese) [inline core/src/test/scala/au/com/cba/omnia/omnitool/ValidatedSpec.scala]: Why did we remove these tests?

Comment (stephanh) [inline core/src/test/scala/au/com/cba/omnia/omnitool/ValidatedSpec.scala]: That was a mistake. Well spotted!

**Comment by vineethvarghese** [comment]
Except for the 2 things mentioned above this looks fine :+1:

File: core/src/main/scala/au/com/cba/omnia/omnitool/Closeable.scala (hunk 1 above)
Comment (stephanh) [inline core/src/main/scala/au/com/cba/omnia/omnitool/Closeable.scala]: You are right. I will fix that.

**Pull request merged**